vms/platformvm/txs/executor/standard_tx_executor.go

```diff
@@ -805,6 +810,146 @@ func (e *StandardTxExecutor) RegisterSubnetValidatorTx(tx *txs.RegisterSubnetVal
+	if err != nil {
+		return err
+	}
+
+	if err := e.Backend.FlowChecker.VerifySpend(
+		tx,
+		e.State,
+		tx.Ins,
+		tx.Outs,
+		e.Tx.Creds,
+		map[ids.ID]uint64{
+			e.Ctx.AVAXAssetID: fee,
+		},
+	); err != nil {
+		return err
+	}
+
+	warpMessage, err := warp.ParseMessage(tx.Message)
+	if err != nil {
+		return err
+	}
+	if warpMessage.NetworkID != e.Ctx.NetworkID {
+		return fmt.Errorf("expected networkID %d but got %d", e.Ctx.NetworkID, warpMessage.NetworkID)
+	}
+
+	addressedCall, err := payload.ParseAddressedCall(warpMessage.Payload)
+	if err != nil {
+		return err
+	}
+
+	msg, err := message.ParseSubnetValidatorWeight(addressedCall.Payload)
+	if err != nil {
+		return err
+	}
+	if err := msg.Verify(); err != nil {
+		return err
+	}
+
+	sov, err := e.State.GetSubnetOnlyValidator(msg.ValidationID)
+	if err != nil {
+		return err
+	}
+	if msg.Nonce < sov.MinNonce {
+		return fmt.Errorf("expected nonce to be at least %d but got %d", sov.MinNonce, msg.Nonce)
+	}
+
+	subnetConversion, err := e.State.GetSubnetConversion(sov.SubnetID)
+	if err != nil {
+		return err
+	}
+	if warpMessage.SourceChainID != subnetConversion.ChainID {
+		return fmt.Errorf("expected chainID %s but got %s", subnetConversion.ChainID, warpMessage.SourceChainID)
+	}
+	if !bytes.Equal(addressedCall.SourceAddress, subnetConversion.Addr) {
+		return fmt.Errorf("expected address %s but got %s", subnetConversion.Addr, addressedCall.SourceAddress)
+	}
+
+	txID := e.Tx.ID()
+
+	// We are removing the validator
+	if msg.Weight == 0 {
+		weight, err := e.State.WeightOfSubnetOnlyValidators(sov.SubnetID)
+		if err != nil {
+			return err
+		}
+		if weight == sov.Weight {
+			return errRemovingLastValidator
+		}
+
+		// The validator is currently active, we need to refund the remaining
+		// balance.
+		if sov.EndAccumulatedFee != 0 {
+			var remainingBalanceOwner message.PChainOwner
+			if _, err := txs.Codec.Unmarshal(sov.RemainingBalanceOwner, &remainingBalanceOwner); err != nil {
+				return err
+			}
+
+			accruedFees := e.State.GetAccruedFees()
+			if sov.EndAccumulatedFee <= accruedFees {
+				// This check should be unreachable. However, including it ensures
+				// that AVAX can't get minted out of thin air due to state
+				// corruption.
+				return fmt.Errorf("%w: validator should have already been disabled", errStateCorruption)
+			}
+			remainingBalance := sov.EndAccumulatedFee - accruedFees
+
+			utxo := &avax.UTXO{
+				UTXOID: avax.UTXOID{
+					TxID:        txID,
+					OutputIndex: uint32(len(tx.Outs)),
+				},
+				Asset: avax.Asset{
+					ID: e.Ctx.AVAXAssetID,
+				},
+				Out: &secp256k1fx.TransferOutput{
+					Amt: remainingBalance,
+					OutputOwners: secp256k1fx.OutputOwners{
+						Threshold: remainingBalanceOwner.Threshold,
+						Addrs:     remainingBalanceOwner.Addresses,
+					},
+				},
+			}
+			e.State.AddUTXO(utxo)
+		}
+	}
+
+	// If the weight is being set to 0, the validator is being removed and the
+	// nonce doesn't matter.
```

Comment:
```suggestion
	// If the weight is being set to 0, the validator is being removed and the
	// nonce doesn't matter. If not, msg.Nonce is less than MaxUInt64 and
	// can be incremented without overflow.
```

Reply:
do we actually need that comment at all?

Reply:
I think the comment is important to make it clear that the potential overflow is not a problem.

Reply:
but we still increment it regardless, so it looks a bit like comment is not in sync with the code. Maybe we should emphasize that it does not matter since it's handled in PutSubnetOnlyValidator.

Reply:
Sorry I'm not sure why the comment seems to be out of sync with the code. I'm expanding the comment to be:
```golang
// If the weight is being set to 0, the validator is being removed and the
// nonce doesn't matter. If not, [msg.Nonce] is enforced by [msg.Verify()]
// to be less than MaxUInt64 and can therefore be incremented without
// overflow.
```
Does that make sense?

Reply:
Changed to be:
```golang
// If the weight is being set to 0, it is possible for the nonce increment
// to overflow. However, the validator is being removed and the nonce
// doesn't matter. If weight is not 0, [msg.Nonce] is enforced by
// [msg.Verify()] to be less than MaxUInt64 and can therefore be incremented
// without overflow.
```
to be explicit about the overflow happening.